Features/ErrorViews/ErrorView.swift

```diff
@@ -0,0 +1,106 @@
+    }()
+    
+    lazy var errorLabel: UILabel = {
+       let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = UIFont.systemFont(ofSize: 15)
+        label.textColor = UIColor(named: "errorTextColor")
+        label.textAlignment = .center
+        label.numberOfLines = 0
+        return label
+    }()
+    
+    lazy var dismissButton: UIButton = {
+       let button = UIButton()
+        let attributes: [NSAttributedString.Key : Any] = [
+            .foregroundColor : UIColor(named: "dismissButtonColor")!,
+            .font : UIFont.systemFont(ofSize: 15)
+        ]
+        let title = NSAttributedString(string: "Dismiss", attributes: attributes)
+        button.setAttributedTitle(title, for: .normal)
+        button.addTarget(self, action: #selector(dismissController), for: .touchUpInside)
+        return button
+    }()
+    
+    lazy var stackView: UIStackView = {
+        let view = UIStackView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.axis = .vertical
+        view.spacing = 24
+        view.alignment = .center
+        view.layoutMargins = UIEdgeInsets(top: 24, left: 32, bottom: 16, right: 32)
+        view.isLayoutMarginsRelativeArrangement = true
+        view.backgroundColor = UIColor(named: "errorViewBackgroundColor")
+        view.layer.cornerRadius = 5
+        view.layer.masksToBounds = true
+        return view
+    }()
+    
+    lazy var buttonStackView: UIStackView = {
+        let view = UIStackView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.axis = .vertical
+        view.alignment = .trailing
+        return view
+    }()
+    
+    weak var delegate: ErrorViewDelegate?
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        setupUI()
+    }
+    
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        
+        setupUI()
+    }
+    
+    private func setupUI() {
+//        At this point I'd talk to the designer about how cruial the background with 0.6 opacity is.
+        backgroundColor = UIColor.clear
+        
+        addSubview(stackView)
+        NSLayoutConstraint.activate([
+            stackView.centerXAnchor.constraint(equalTo: safeAreaLayoutGuide.centerXAnchor),
+            stackView.centerYAnchor.constraint(equalTo: safeAreaLayoutGuide.centerYAnchor),
+            stackView.widthAnchor.constraint(equalTo: safeAreaLayoutGuide.widthAnchor, multiplier: 0.76),
+            stackView.heightAnchor.constraint(lessThanOrEqualTo: safeAreaLayoutGuide.heightAnchor, multiplier: 0.8)
+        ])
+        
+        stackView.addArrangedSubview(imageView)
+        stackView.addArrangedSubview(errorLabel)
+        stackView.addArrangedSubview(buttonStackView)
+        NSLayoutConstraint.activate([
+            buttonStackView.trailingAnchor.constraint(equalTo: stackView.layoutMarginsGuide.trailingAnchor)
+        ])
```

Comment:
When using stack views try avoiding setting constraints to views that are inside of the stack, this can go to some weird directions sometimes. Stacks can be super annoying to deal so what to do here: 

1. First, usually you will always set your stack alignment to `.fill`, specially on high level stacks. So in our case here we need to set the high level stack alignment to `.fill`.

2. Secondly, our image view will try to fill it ignoring aspect ratio, since we need it centered we make `imageView.contentMode = .center`

3. Lastly, your approach for positioning the button is super on point! Already prepared to handle multiple buttons as well so congrats! We just need to achieve this behaviour (pushing to the right) without a constraint. Since our main stack is now `.fill` mode, the button stack will fill the full space, to move the button to the right most we simply add a spacer to the beginning. In this case a simple UIView can do the trick since buttons by default hug themselves to the minimum space already, the UIView will simply take as much space available as possible pushing the button all the way to the right. Just don't forget to update the buttons stack axis to `.horizontal`, alignment to `.fill`, and remove the manual constraints

Reply:
Beautiful, thanks!
Stacks are tricky beasts :)